### PR TITLE
feat(vpn): VPN gateway flavor field supports updates

### PR DESCRIPTION
**Pull request by profoundwu**
**What this PR does / why we need it**:
VPN gateway flavor field supports updates
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. VPN gateway flavor field supports updates
```

## PR Checklist

* [X] Tests added/passed.

```
 make testacc TEST="./huaweicloud/services/acceptance/vpn" TESTARGS="-run TestAccGateway_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/vpn -v -run TestAccGateway_basic -timeout 360m -parallel 4
=== RUN   TestAccGateway_basic
=== PAUSE TestAccGateway_basic
=== CONT  TestAccGateway_basic
--- PASS: TestAccGateway_basic (467.68s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/vpn       467.741s

make testacc TEST="./huaweicloud/services/acceptance/vpn" TESTARGS="-run TestAccGateway_UpdateWithEpsId"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/vpn -v -run TestAccGateway_UpdateWithEpsId -timeout 360m -parallel 4
=== RUN   TestAccGateway_UpdateWithEpsId
=== PAUSE TestAccGateway_UpdateWithEpsId
=== CONT  TestAccGateway_UpdateWithEpsId
--- PASS: TestAccGateway_UpdateWithEpsId (433.17s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/vpn       433.247s

```

* [X] Documentation updated.
* [X] Schema updated.
* [ ] CheckDeleted.

  - **a. During query operation (Read Context)**
    aa. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<

    
  - **b. During delete/disassociate/unbind operation (Delete Context)**
    ba. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<

### CODE REVIEW

**Comment by Alice1319** [comment]
LGTM

**Comment by WangLiNaruto** [comment]
LGTM

**Comment by yuanxiaojun987** [comment]
LGTM

File: huaweicloud/services/vpn/resource_huaweicloud_vpn_gateway.go [hunk 1]
```diff
@@ -71,7 +72,6 @@ func ResourceGateway() *schema.Resource {
 			"flavor": {
 				Type:        schema.TypeString,
 				Optional:    true,
-				ForceNew:    true,
 				Computed:    true,
 				Description: `The flavor of the VPN gateway.`,
 				ValidateFunc: validation.StringInSlice([]string{
```
Comment (houpeng80) [inline huaweicloud/services/vpn/resource_huaweicloud_vpn_gateway.go:77]: remove the validation? the value of flavor can not be V1G or V300

**Pull request merged**